openslides/core/static/js/core/site.js

```diff
@@ -908,205 +896,45 @@ angular.module('OpenSlidesApp.core.site', [
-                        var data = {};
-                        // calculate rest duration of countdown (in seconds!)
-                        var newDuration = Math.floor( countdown.countdown_time - Date.now() / 1000 + $scope.serverOffset );
-                        data[uuid] = {
-                            'running': false,
-                            'countdown_time': newDuration
-                        };
-                        $http.post('/rest/core/projector/' + projector.id + '/update_elements/', data);
-                    }
-                });
-            });
-        };
-        $scope.resetCountdown = function (countdown) {
-            $scope.projectors.forEach(function (projector) {
-                _.forEach(projector.elements, function (element, uuid) {
-                    if (element.name == 'core/countdown' && element.index == countdown.index) {
-                        var data = {};
-                        data[uuid] = {
-                            'running': false,
-                            'countdown_time': countdown.default_time,
-                        };
-                        $http.post('/rest/core/projector/' + projector.id + '/update_elements/', data);
-                    }
-                });
+            var isProjectedIds = countdown.isProjected();
+            _.forEach(isProjectedIds, function(id) {
+                countdown.project(id);
             });
+            Countdown.destroy(countdown.id);
         };
 
         // *** message functions ***
         $scope.editMessage = function (message) {
             message.editFlag = false;
-            $scope.projectors.forEach(function (projector) {
-                _.forEach(projector.elements, function (element, uuid) {
-                    if (element.name == 'core/message' && element.index == message.index) {
-                        var data = {};
-                        data[uuid] = {
-                            message: message.message,
-                        };
-                        $http.post('/rest/core/projector/' + projector.id + '/update_elements/', data);
-                    }
-                });
-            });
+            ProjectorMessage.save(message);
         };
         $scope.addMessage = function () {
-            $scope.highestMessageIndex++;
-            // select all projectors on creation, so write the countdown to all projectors
-            $scope.projectors.forEach(function (projector) {
-                $http.post('/rest/core/projector/' + projector.id + '/activate_elements/', [{
-                    name: 'core/message',
-                    visible: false,
-                    selected: true,
-                    index: $scope.highestMessageIndex,
-                    message: '',
-                    stable: true,
-                }]);
-            });
+            var message = {message: gettextCatalog.getString('Message')};
```

Comment:
Why do you create a default text for a new message? - Now you have to remove this default text first to add custom message text. I prefere to remove this default text and create an empty message.